src/components/inline-rendered-diff.jsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+
+/**
+ * @typedef {Object} InlineRenderedDiffProps
+ * @property {Diff} diff The diff to render
+ * @property {Page} page The page this diff pertains to
+ */
+
+/**
+ * Display two versions of a page with their changes inline together.
+ *
+ * @class InlineRenderedDiff
+ * @extends {React.Component}
+ * @params {InlineRenderedDiffProps} props
+ */
+export default class InlineRenderedDiff extends React.Component {
+  constructor (props) {
+    super(props);
+    this.frame = null;
+  }
+
+  render () {
+    if (!this.props) {
+      return null;
+    }
+
+    return (
+      <div className="inline-render">
+        <iframe sandbox="allow-forms allow-scripts" ref={frame => this.frame = frame} />
+      </div>
+    );
+  }
+
+  /**
+     * @param {InlineRenderedDiffProps} nextProps
+     * @param {Object} nextState
+     * @returns {boolean}
+     */
+  shouldComponentUpdate (nextProps, nextState) {
+    return nextProps.diff.from_version_id !== this.props.diff.from_version_id
+      || nextProps.diff.to_version_id !== this.props.diff.to_version_id;
+  }
+
+  componentDidMount () {
+    this._updateContent();
+  }
+
+  componentDidUpdate () {
+    this._updateContent();
+  }
+
+  _updateContent () {
+    const raw_source = this.props.diff.content.diff;
+
+    this.frame.setAttribute(
+      'srcdoc',
+      renderableSource(raw_source, this.props.page));
+  }
+}
+
+/**
+ * Create renderable HTML source code from the raw HTML source of the diff.
+ *
+ * @param {string} source Full diff source code
+ * @param {Page} page The page that is being diffed
+ * @param {string} viewType Either `additions` or `removals`
+ */
+function renderableSource (source, page, viewType) {
+  const parser = new DOMParser();
+  const newDocument = parser.parseFromString(source, 'text/html');
+  renderableDocument(newDocument, page);
+  const prefix = source.match(/^[^]*?<html/ig)[0];
+  return prefix + newDocument.documentElement.outerHTML.slice(5);
+}
+
+/**
+ * Process HTML document so that it renders nicely. This includes things like
+ * adding a `<base>` tag so subresources are properly fetched.
+ *
+ * @param {HTMLDocument} sourceDocument
+ * @param {Page} page
+ * @returns {HTMLDocument}
+ */
+function renderableDocument (sourceDocument, page) {
```

Comment:
Because `renderableDocument()` is the same function as in `side-by-side-rendered-diff`, I wonder if we can combine `side-by-side` and `inline` into a new component called something like `rendered-item` that returns a single iframe and determines how to manipulate it's source using a new input of `viewType` in props. 

`side-by-side-rendered-diff` and `inline-rendered-diff` could then return the correct number and type of `rendered-items` the same way that `changes-only-diff` and `highlighted-text-diff` use `diff-item`. Not sure if the iframes would update correctly in this case though. ¯\_(ツ)_/¯ Haven't worked with refs before, but I keep reading about how fidgety they can be.

If not, is there a good way for the views to share this function? Pulling this out into a separate file seems overkill, but having something that does this much work in two places is concerning.

Reply:
Oops, though I’d replied to this before… sure. I didn’t want to do this because it would cause massive conflicts with your work on #156.

I don’t think the specific method you’re describing above would work, but `SideBySideRenderedDiff` could easily be composed of two `InlineRenderedDiff` components or of two of some new iframe-y component that does 90% of what `InlineRenderedDiff` does (and `InlineRenderedDiff` would use it, too).

Reply:
The conflicts shouldn't a problem. As you mentioned in the review, what I was trying to do with the rendered views wasn't ideal anyway and prompted this change. 

Luckily, I've been learning to keep my commits better separated. I can easily revert back one commit where I was working on the rendered views, rebase on top of this, and be about my merry way.

Feel free to compose the the rendered diffs whichever seems best and I'll incorporate it into #156.